src/rest/app.py

```diff
@@ -0,0 +1,178 @@
+async def get_transcriptions_id(transcription_id: str):
+    """Get the status of a transcription by ID."""
+    file = DATA_HANDLER.get_status_file_by_id(transcription_id)
+    if file:
+        return JSONResponse(content=file, status_code=200)
+    raise HTTPException(status_code=404, detail="Transcription ID not found")
+
+
+@app.post("/transcriptions", dependencies=[Depends(require_api_key)])
+async def post_transcription(
+    file: UploadFile = File(...),
+    language: str | None = Form(None),
+    settings: str | None = Form(None),
+    model: str | None = Form(None),
+    task: str = Form("transcribe"),
+    text: str | None = Form(None),
+):
+    """Transcribe an audio file."""
+    if language and language not in config["supported_language_codes"]:
+        raise HTTPException(
+            status_code=400,
+            detail=f"Unsupported language code: {language}",
+        )
+
+    transcription_id = str(uuid.uuid4())
+    try:
+        audio = AudioSegment.from_file(file.file)
+        result = DATA_HANDLER.save_audio_file(audio, transcription_id)
+        if not result["success"]:
+            raise HTTPException(status_code=400, detail=result["message"])
+    except Exception as e:
+        LOGGER.error(f"Audio processing error: {e}")
+        raise HTTPException(status_code=500, detail="Something went wrong")
+
+    time.sleep(0.1)  # ensure file is saved
+
+    if task == "align" and (not text or not language):
+        missing_field = "text" if not text else "language"
+        raise HTTPException(
+            status_code=400,
+            detail=f"property {missing_field} is required for task align"
+        )
+
+    settings_dict = json.loads(settings) if settings else None
+    data = TranscriptionData(
+        transcription_id=transcription_id,
+        status=TranscriptionStatus.IN_QUERY.value,
+        start_time=datetime.utcnow().isoformat() + 'Z',
+        settings=settings_dict,
+        model=model,
+        task=task,
+        text=text,
+        language=language,
+    ).model_dump()
+
+    DATA_HANDLER.write_status_file(transcription_id, data)
+    return JSONResponse(content=data, status_code=200)
+
+
+@app.get("/export/transcript/{transcription_id}", dependencies=[Depends(require_api_key)])
+async def get_stream_transcript_export(transcription_id: str):
+    """Get the transcription JSON for a specific ID."""
+    file = DATA_HANDLER.get_export_json_by_id(transcription_id)
+    if file:
+        return JSONResponse(content=file, status_code=200)
+    raise HTTPException(status_code=404, detail="Transcription ID not found")
+
+
+@app.get("/export/audio/{transcription_id}", dependencies=[Depends(require_api_key)])
+async def get_stream_audio_export(transcription_id: str):
+    """Get the audio WAV file for a specific transcription ID."""
+    file = DATA_HANDLER.get_audio_file_by_id(transcription_id)
+    if file is not None:
```

Comment:
I think its prettier to just flip this.
Keyword: Guard Clauses

Reply:
tbh, I had it flipped at first and found it ugly hahhaha